crates/ignore/tests/bom_test.rs

```diff
@@ -0,0 +1,23 @@
+use std::path::Path;
+
+use ignore::gitignore::{Gitignore, GitignoreBuilder};
+
+const IGNORE_FILE: &'static str = "tests/bom_test.gitignore";
+
+fn get_gitignore() -> Gitignore {
+    let mut builder = GitignoreBuilder::new("ROOT");
+    let error = builder.add(IGNORE_FILE);
```

Comment:
I added this as an integration test because the existing unit tests in gitignore.rs don't create any files but the logic to test is in `GitignoreBuilder.add` which only works on files.

I could also
1. either add a unit test that creates a temporary gitignore file,
2. or extract a method from `add` that works on a `Read` and test that. However, since the errors need the path, the extracted method would also need to know the path.